hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/HoodieSparkIndexClient.java

```diff
@@ -86,9 +92,45 @@ public static HoodieSparkIndexClient getInstance(SparkSession sparkSession) {
 
   @Override
   public void create(HoodieTableMetaClient metaClient, String userIndexName, String indexType, Map<String, Map<String, String>> columns, Map<String, String> options) throws Exception {
+    if (indexType.equals(PARTITION_NAME_SECONDARY_INDEX) || indexType.equals(PARTITION_NAME_BLOOM_FILTERS)
+        || indexType.equals(PARTITION_NAME_COLUMN_STATS)) {
+      createGenericIndex(metaClient, userIndexName, indexType, columns, options);
+    } else {
+      createRecordIndex(metaClient, userIndexName, indexType);
```

Comment:
Can we mock and add a UT for this client just to validate the conditional logics?

Reply:
Yes, all the error cases are handled already.
Record index related scenarios in TestIndexSyntax(Test Create and Drop Index Syntax with Simple record key, Test Create and Drop Index Syntax with Complex record key)

Secondary Index - 
TestSecondaryIndex(Test Create/Show/Drop Secondary Index)

Expression Index
TestExpressionIndex(Test Create Expression Index, testBloomFiltersIndexPruning)